src/DictVectors/stochasticstyle.jl

```diff
@@ -1,73 +1,57 @@
 """
     StochasticStyle(v)
-    StochasticStyle(typeof(v))
-`StochasticStyle` specifies the native style of the generalised vector `v` that
-determines how simulations are to proceed. This can be fully stochastic (with
-`IsStochastic`), fully deterministic (with `IsDeterministic`), or stochastic with
-floating point walker numbers and threshold (with [`IsStochasticWithThreshold`](@ref)).
-"""
-abstract type StochasticStyle end
-
-# some sensible defaults
-StochasticStyle(A::Union{AbstractArray,AbstractDVec}) = StochasticStyle(typeof(A))
-StochasticStyle(::Type{<:Array}) = IsDeterministic()
-StochasticStyle(::Type{Vector{Int}}) = IsStochastic()
-function StochasticStyle(T::Type{<:AbstractDVec{K,V}}) where {K,V<:AbstractFloat}
-    IsDeterministic()
-end
-function StochasticStyle(T::Type{<:AbstractDVec{K,V}}) where {I<:Integer,K,V<:Complex{I}}
-    IsStochastic2Pop()
-end
-function StochasticStyle(T::Type{<:AbstractDVec})
-    IsStochastic()
-end
+`StochasticStyle` specifies the native style of the generalised vector `v` that determines
+how simulations are to proceed. This can be fully stochastic (with `IsStochasticInteger`),
+fully deterministic (with `IsDeterministic`), or stochastic with floating point walker
+numbers and threshold (with [`IsStochasticWithThreshold`](@ref)).
 
+When defining a new `StochasticStyle`, subtype it as `MyStyle<:StochasticStyle{T}` where `T`
+is the concrete value type the style is designed to work with.
 """
-    IsStochastic()
+abstract type StochasticStyle{T} end
+
+Base.eltype(::Type{<:StochasticStyle{T}}) where {T} = T
+
+"""
+    StyleUnknown()
+Trait for value types not (currently) compatible with FCIQMC. This style makes it possible to
+construct dict vectors with unsupported `valtype`s.
+"""
```

Comment:
Why do we need this?

Currently it will not work with the call signature as per docstring. If we need it, maybe replace the docstring by
```suggestion
"""
    StyleUnknown{T}() <:StochasticStyle{T}
Trait for value types not (currently) compatible with FCIQMC. This style makes it possible to
construct dict vectors with unsupported `valtype`s. See [`StochasticStyle`](@ref).
"""
```
In general, does it make sense to annotate in the docstring the abstract supertype? It is not part of the call signature, obviously, but I think that it would help user who reads the docstring to better understand that this is one way to generate a `StochasticStyle`. 

In addition, or as an alternative, we could add a link to the docstring of `StochasticStyle`.

I would prefer to have both. If we agree on this, I would suggest to add it to all constructors for `StochasticStyles`.